sdk/include/opentelemetry/sdk/resource/resource_detector.h

```diff
@@ -39,6 +39,17 @@ class OTELResourceDetector : public ResourceDetector
   Resource Detect() noexcept override;
 };
 
+/**
+ * ContainerResourceDetector to detect resource attributes when running inside a containerized
+ * environment. This detector extracts metadata such as container ID from cgroup information and
+ * sets attributes like container.id following the OpenTelemetry semantic conventions.
+ */
+class ContainerResourceDetector : public ResourceDetector
+{
+public:
+  Resource Detect() noexcept override;
+};
+
```

Comment:
`ContainerResourceDetector `should be under separate file, and content of common/container.h should also be within that?